server/src/main/java/com/genymobile/scrcpy/ScreenEncoder.java

```diff
@@ -148,15 +167,20 @@ private boolean encode(MediaCodec codec, FileDescriptor fd) throws IOException {
         boolean eof = false;
         MediaCodec.BufferInfo bufferInfo = new MediaCodec.BufferInfo();
 
-        while (!consumeRotationChange() && !eof) {
-            int outputBufferId = codec.dequeueOutputBuffer(bufferInfo, -1);
+        while (!consumeRotationChange() && !consumeKeyFrameRequest() && !eof) {
+            // NOTE(frankleonrose): Timeout after 10ms to respond to key frame request.
+            int outputBufferId = codec.dequeueOutputBuffer(bufferInfo, 10000);
```

Comment:
Just want to double-check that this is measured in microseconds and is actually 10ms of time.

Reply:
Yes, µs `timeoutUs` from [the docs](https://developer.android.com/reference/android/media/MediaCodec#dequeueOutputBuffer(android.media.MediaCodec.BufferInfo,%20long)).